src/command_modules/azure-cli-eventgrid/azure/cli/command_modules/eventgrid/custom.py

```diff
@@ -284,7 +392,160 @@ def update_event_subscription(
+    # Supplied endpoint would be in the following format:
+    # /subscriptions/.../storageAccounts/sa1/queueServices/default/queues/{queueName}))
+    # and we need to break it up into:
+    # /subscriptions/.../storageAccounts/sa1 and queueName
+    queue_items = re.split(
+        "/queueServices/default/queues/", endpoint, flags=re.IGNORECASE)
+
+    if len(queue_items) != 2 or queue_items[0] is None or queue_items[1] is None:
+        raise CLIError('Argument Error: Expected format of --endpoint for storage queue is:' +
+                       '/subscriptions/id/resourceGroups/rg/providers/Microsoft.Storage/' +
+                       'storageAccounts/sa1/queueServices/default/queues/queueName')
+
+    destination = StorageQueueEventSubscriptionDestination(
+        resource_id=queue_items[0], queue_name=queue_items[1])
+
+    return destination
+
+
+def _get_deadletter_destination(deadletter_endpoint):
+    blob_items = re.split(
+        "/blobServices/default/containers/", deadletter_endpoint, flags=re.IGNORECASE)
+
+    if len(blob_items) != 2 or blob_items[0] is None or blob_items[1] is None:
+        raise CLIError('Argument Error: Expected format of --deadletter-endpoint is:' +
+                       '/subscriptions/id/resourceGroups/rg/providers/Microsoft.Storage/' +
+                       'storageAccounts/sa1/blobServices/default/containers/containerName')
+
+    deadletter_destination = StorageBlobDeadLetterDestination(
+        resource_id=blob_items[0], blob_container_name=blob_items[1])
+
+    return deadletter_destination
+
+
+def _validate_retry_policy(max_delivery_attempts, event_ttl):
+    if max_delivery_attempts < 1 or max_delivery_attempts > 30:
+        raise CLIError('--max-delivery-attempts should be a number between 1 and 30.')
+
+    if event_ttl < 1 or event_ttl > 1440:
+        raise CLIError('--event-ttl should be a number between 1 and 1440.')
+
+
+def _warn_if_manual_handshake_needed(endpoint_type, endpoint):
+    # If the endpoint belongs to a service that we know implements the subscription validation
+    # handshake, there's no need to show this message, hence we check for those services
+    # before showing this message. This list includes Azure Automation, EventGrid Trigger based
+    # Azure functions, and Azure Logic Apps.
+    if endpoint_type.lower() == WEBHOOK_DESTINATION.lower() and \
+       "azure-automation" not in endpoint.lower() and \
+       "eventgridextension" not in endpoint.lower() and \
+       "logic.azure" not in endpoint.lower():
+
+        logger.warning('If the provided endpoint does not support subscription validation '
+                       'handshake, navigate to the validation URL that you receive in the '
+                       'subscription validation event, in order to complete the event '
+                       'subscription creation or update. For more details, '
+                       'please visit http://aka.ms/esvalidation')
```

Comment:
You can use a validator for this.

Reply:
Opened issue #8213 to handle all validator work separately.